### DISCUSSION TITLE
Adding attachments to allure report via direct addition to junit xml file

### DISCUSSION THREAD

**Discussion by raj-nimble** [allure-framework/allure2#2651] [General Discussion]
Hi, we have a tool written in rust that is currently writing out test results as junit.xml. We then use allure generate on these xml results.
The problem I am having is that I cannot figure out how to add attchements/screenshots via this method. Is this possible to do? We can output in a different common format if needed, but it doesn't seem practical to write out the large amount of allure format json files in lieu of the single file per test suite we currently generate.

I did attempt using the linked convention, where the attachment is a property for the test case, but that didn't work.
https://github.com/testmoapp/junitxml?tab=readme-ov-file#attachments
https://github.com/testmoapp/junitxml?tab=readme-ov-file#conventions-attachments

Is there a good solution for this? Maybe supporting the above convention could be added as a feature?

**Comment by baev** [comment]
The issue with `junit.xml` is that it is undocumented and has multiple dialects, each with custom attributes, different attribute types, and formats. By default, `junit.xml` doesn't support adding test attachments. Suite logs can be added by creating a file with the same name but with a .log extension. These logs are shared across all the tests within a result file.

The link you provided refers to a specific dialect of `junit.xml` used by the Testmo teIt'sIt’s just a naming convention supported by their tool. I am unaware of any test frameworks that produce such an output, and I don't see any reason to introduce support for this naming convention in Allure.

I suggest using the Allure results format for all custom integrations: https://allurereport.org/docs/how-it-works-test-result-file/

Allure supports steps (including nesting), attachments (for tests and steps), parameters, custom metadata, and more. It is simple to use: there is only one required field — the test name (though you will likely need to specify the status as well):

```json
{
  "name": "A passed test result",
  "status": "passed"
}
```

Allure requires creating a separate result file for each test, which is intentional. The results are stored on the file system as soon as a test finishes, making the system more failure-tolerant. Even if the test process fails, you will still have access to the results of the already completed tests. Consequently, you can have real-time reporting (already supported by Allure TestOps and will soon be included in the Allure Report). Additionally, it provides first-class support for distributed test execution.

**Comment by raj-nimble** [comment]
Ok thanks, will explore more closely using allures results format.

**Comment by aburns14** [comment]
I would love to reopen this discussion or get pointed to a way to update report meta data to point to generated screenshots our framework produces that is not compatible.  There has to be a way to update allure to inject screenshot meta data post generation. or at least provide instructions before an allure generate is called.

**Reply by baev** [reply]
Please create a separate discussion instead
